bigquery_etl/stripe/__init__.py

```diff
@@ -141,56 +195,113 @@ def schema(resource: Type[ListableAPIResource]):
 )
 @click.option(
     "--resource",
-    default="Event",
     type=StripeResourceType(),
     help="Type of stripe resource to import",
 )
+@click.option(
+    "--report-type",
+    help="Stripe report type to import",
+)
+@click.option(
+    "--time-partitioning-field",
+    default="created",
+    help="Field to use for partitioning and clustering; if --date or --before-date or"
+    " --after-date are specified, values must fall within that window",
+)
+@click.option(
+    "--time-partitioning-type",
+    default=bigquery.TimePartitioningType.DAY,
+    type=click.Choice(
+        [bigquery.TimePartitioningType.DAY, bigquery.TimePartitioningType.MONTH]
+    ),
+    help="BigQuery time partitioning type for --table",
+)
 def stripe_import(
     api_key: Optional[str],
     date: Optional[datetime],
+    after_date: Optional[datetime],
     before_date: Optional[datetime],
     table: Optional[str],
     format_resources: bool,
     strict_schema: bool,
     quiet: bool,
-    resource: Type[ListableAPIResource],
+    resource: Optional[Type[ListableAPIResource]],
+    report_type: Optional[str],
+    time_partitioning_field: str,
+    time_partitioning_type: str,
 ):
     """Import Stripe data into BigQuery."""
+    if after_date:
+        after_date = after_date.replace(tzinfo=timezone.utc)
+    if before_date:
+        before_date = before_date.replace(tzinfo=timezone.utc)
+    if date:
+        date = date.replace(tzinfo=timezone.utc)
+        if time_partitioning_type == bigquery.TimePartitioningType.DAY:
+            after_date = date
+            before_date = date + relativedelta(days=1)
+            if table:
+                table = f"{table}${date:%Y%m%d}"
+        elif time_partitioning_type == bigquery.TimePartitioningType.MONTH:
+            after_date = date.replace(day=1)
+            before_date = date + relativedelta(months=1)
+            if table:
+                table = f"{table}${date:%Y%m}"
     if resource is stripe.Event and not date:
-        click.echo("must specify --date for --resource=Event")
-        sys.exit(1)
-    if table and date:
-        table = f"{table}${date:%Y%m%d}"
+        raise click.ClickException("must specify --date for --resource=Event")
+    if report_type and not date:
+        raise click.ClickException("must specify --date for --report-type")
```

Comment:
While the `--after-date` and `--before-date` options don't appear to currently be used, these checks seem to preclude them from ever being used.